electrum/coinchooser.py

```diff
@@ -474,15 +474,54 @@ def penalty(buckets: List[Bucket]) -> ScoredCandidate:
         return penalty
 
 
+class CoinChooserMinChange(CoinChooserRandom):
+    """Attempts to create a transaction with the smallest fee. Created as a coin chooser for alert transactions, with
+    the main goal of reducing the amount of transaction change, which is locked for 24h.
+    First, every bucket is a single utxo.
+    Second, it penalizes multiple number of inputs
+    Third, it proportionally penalizes every change greater than the minimal.
+    """
+
+    def keys(self, coins):
+        # Give a random indentifier to each bucket.
+        return [os.urandom(16).hex() for _ in coins]
+
+    def penalty_func(self, base_tx, *, tx_from_buckets):
+        min_output = min(o.value for o in base_tx.outputs())
+
+        def penalty(buckets: List[Bucket]) -> ScoredCandidate:
+            # Penalize using many buckets (~inputs)
+            badness = len(buckets) - 1
+            tx, change_outputs = tx_from_buckets(buckets)
+            change = sum(o.value for o in change_outputs)
+            # Penalize change not roughly in output range
+            if change == 0:
+                pass  # no change is great!
```

Comment:
`pass` does nothing so the `if change == 0` does not have any effect.
did you mean smth else?
in that case, `change == 0` is also `< COIN / 1000` so the badness will be increased